torchaudio/functional/functional.py

```diff
@@ -651,6 +652,8 @@ def mu_law_decoding(
     mu = quantization_channels - 1.0
     if not x_mu.is_floating_point():
         x_mu = x_mu.to(torch.float)
+    else:
+        warnings.warn("The input Tensor must be of int type. This will be an error in the v0.12 release.", FutureWarning)
```

Comment:
Please note this is different than the suggestion in the issue, so please let me know whether or not my assumption that in the decode the expected input is `int` is correct.

Reply:
In case of decoding, I think both `int` and `float` types should be allowed, as both of them can handle the expected value range.

The issue in encoding is that, the input Tensor has to be normalized to have values in range of `[-1, 1]`. Integer tensor can only express three values. (`-1`, `0`, and `1`) in this range, which is rarely a correct input.